src/offset_of.rs

```diff
@@ -67,6 +67,28 @@ macro_rules! _memoffset_offset_from_unsafe {
         ($field as usize) - ($base as usize)
     };
 }
+#[cfg(not(feature = "unstable_offset_of"))]
+#[macro_export(local_inner_macros)]
+#[doc(hidden)]
+macro_rules! _memoffset__offset_of_impl {
+    ($parent:path, $field:tt) => {{
+        // Get a base pointer (non-dangling if rustc supports `MaybeUninit`).
+        _memoffset__let_base_ptr!(base_ptr, $parent);
+        // Get field pointer.
+        let field_ptr = raw_field!(base_ptr, $parent, $field);
+        // Compute offset.
+        _memoffset_offset_from_unsafe!(field_ptr, base_ptr)
+    }};
+}
+#[cfg(feature = "unstable_offset_of")]
+#[macro_export]
+#[doc(hidden)]
+#[allow_internal_unstable(offset_of)]
+macro_rules! _memoffset__offset_of_impl {
+    ($parent:path, $field:tt) => {{
+        $crate::__priv::mem::offset_of!($parent, $field)
+    }};
+}
```

Comment:
What is the point of introducing a separate `_memoffset__offset_of_impl` macro here? Why not directly define `offset_of` under a `cfg` one way or the other?

Reply:
https://github.com/Gilnaa/memoffset/pull/72#discussion_r1195976297

Reply:
Ah, I see. Would be good to have a comment in the code explaining this.

Reply:
Good idea; will do

Reply:
https://github.com/Gilnaa/memoffset/pull/73